packages/project-themes/lektor_project_themes.py

```diff
@@ -0,0 +1,307 @@
+            name = image['path']
+            name = self._get_new_image_name(name)
+            api_url = image['url']
+            image_bytes = self._download_image(api_url)
+            path = os.path.join(path_theme_dir, name)
+            with open(path, 'wb') as f:
+                f.write(image_bytes)
+
+    @staticmethod
+    def _get_new_image_name(name):
+        """
+        Gets the new name for the image
+
+        :param str name: Current name of image file
+        :return: New name of image file
+        :rtype: str
+        """
+        new_names = {
+            'homepage.png': '1 - homepage.png',
+        }
+
+        return new_names.get(name, name)
+
+    @staticmethod
+    def _download_image(api_url):
+        """
+        Download image
+
+        :param str api_url: API URL of image
+        :return: Bytes of image
+        :rtype: bytes
+        """
+        response = api_get(api_url)
+        image_bytes = response['content']
+        image_bytes = base64.b64decode(image_bytes)
+        return image_bytes
+
+
+def api_get(url):
+    """
+    Get API response
+
+    :param str url: URL to get
+    :return: JSON response
+    :rtype: dict
+    """
+    try:
+        github_token = os.environ['GITHUB_TOKEN']
+    except KeyError:
+        raise Exception("Cannot access GitHub API without token. Add env var 'GITHUB_TOKEN'")
+
+    headers = {
+        'Authorization': "Bearer {}".format(github_token),
+        'Cache-Control': "no-cache",
+    }
+    response = requests.get(url, headers=headers)
+    return response.json()
+
+
+class _Theme(object):
+    """A single theme"""
+
+    def __init__(self, repo_name, api_url):
+        """
+        init
+
+        :param str repo_name: Name of repo
+        :param str api_url: API url for repo
+        """
+        self.repo_name = repo_name
+        self.api_url = api_url
+        self.api_response = api_get(api_url)
+        self.listing = self.api_response['tree']  # type: dict
+        self.images = self._load_images()  # type: List[dict]
+
+        self.ini = self._load_ini()  # type: ConfigParser
+        theme_data = self.ini['theme']
+        self.name = theme_data['name']  # type: str
+        self.description = theme_data['description']  # type: str
+        self.homepage = theme_data['homepage']  # type: str
+        author_data = self.ini['author']
+        self.author_name = author_data['name']  # type: str
+        self.author_homepage = author_data['homepage']  # type: str
+
+    def __repr__(self):
+        return "<Theme {}>".format(self.name)
+
+    def _load_images(self):
+        """
+        Loads image info from the API
+
+        :return: Data of images
+        :return: list[dict]
+        """
+        try:
+            images_api = [l for l in self.listing if l['path'] == 'images' and l['type'] == 'tree'][0]
+        except IndexError:
+            raise IndexError("Theme did not contain theme.ini.\n{}".format(self.listing))
```

Comment:
lektor-theme-nix throws this, but actually fails because there are no images, (not no theme.ini).